### PR TITLE
"EX Crystal Guardians" Fixes

### PR DESCRIPTION
**Pull request by starg09**

### CODE REVIEW

File: src/tcgwars/logic/impl/gen3/CrystalGuardians.groovy [hunk 1]
```diff
@@ -1904,7 +1936,7 @@ public enum CrystalGuardians implements LogicCardInfo {
           }
         }
         playRequirement{
-          assert all.findAll {it.cards.hasType(POKEMON_TOOL)} || (bg.stadiumInfoStruct && stadiumCanBeAffectedByItemAndSupporter())
+          assert all.any{it.cards.hasType(POKEMON_TOOL)} || (bg.stadiumInfoStruct && stadiumCanBeAffectedByItemAndSupporter())
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen3/CrystalGuardians.groovy:1939]: Is this trainer at all different than field blower? If not, one should probably copy the other.

File: src/tcgwars/logic/impl/gen3/CrystalGuardians.groovy [hunk 2]
```diff
@@ -779,6 +786,14 @@ public enum CrystalGuardians implements LogicCardInfo {
...
         weakness F
+        pokeBody "Hover Lift", {
+          text "You pay [C] less to retreat your Jigglypuff, Wigglytuff, Wigglytuff ex, and Igglybuff."
+          getterA GET_RETREAT_COST ,{ h->
+            if (h.effect.target.owner == self.owner && ["Jigglypuff", "Wigglytuff", "Wigglytuff ex", "Igglybuff"].contains(h.effect.target.name)) {
+              h.object -= 1
+            }
+          }
+        }
         pokePower "Baby Evolution", {
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen3/CrystalGuardians.groovy:797]: Just making a note here that this Poké-Power isn't on the carddb and should probably be added at some point.

File: src/tcgwars/logic/impl/gen3/CrystalGuardians.groovy [hunk 3]
```diff
@@ -1916,21 +1948,22 @@ public enum CrystalGuardians implements LogicCardInfo {
       case AGGRON_EX_89:
       return evolution (this, from:"Lairon", hp:HP150, type:M, retreatCost:4) {
         weakness R
+        weakness F
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen3/CrystalGuardians.groovy:1951]: Also missing from carddb

File: src/tcgwars/logic/impl/gen3/CrystalGuardians.groovy [hunk 4]
```diff
@@ -2174,14 +2211,17 @@ public enum CrystalGuardians implements LogicCardInfo {
...
+            afterDamage{
+              discardSelfEnergy C,C
+            }
           }
         }
       };
       case SCEPTILE_EX_DELTA_96:
       return evolution (this, from:"Grovyle", hp:HP140, type:P, retreatCost:1) {
         weakness G
+        weakness R
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen3/CrystalGuardians.groovy:2224]: also missing from carddb

File: src/tcgwars/logic/impl/gen3/CrystalGuardians.groovy (hunk 1 above)
Comment (starg09) [reply src/tcgwars/logic/impl/gen3/CrystalGuardians.groovy:1939]: Looking at them, they should be functionally identical. Will use copy on this 👍

**Pull request merged**